src/and_then_filter.rs

```diff
@@ -6,6 +6,50 @@
 
 /// Extension trait for `Iterator<Item = Result<O, E>>` to selectively transform and map Oks and Errors.
 pub trait AndThenFilter<O, E>: Sized {
+    /// Equivalent to [Iterator::filter_map] on all `Ok` values.
+    /// The filter function can fail with a result and turn an
+    /// [Result::Ok] into a [Result::Err]
+    /// ```
+    /// use std::str::FromStr;
+    /// use resiter::and_then_filter::AndThenFilter;
+    ///
+    /// let filter_mapped: Vec<_> = vec![
```

Comment:
I am not entirely sure here, but you might not need `vec![]` here, a normal slice `[...]` should be enough, isn't it?

Reply:
I tried just  to remove the `vec!` but can't really decipher what the compiler tells me when I do that. I will leave this one for now and get back to it.